x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/predicate/conditional/NullIfProcessor.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.sql.expression.predicate.conditional;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xpack.sql.expression.gen.processor.Processor;
+import org.elasticsearch.xpack.sql.expression.predicate.operator.comparison.BinaryComparisonProcessor;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class NullIfProcessor implements Processor {
+
+    public static final String NAME = "nni";
+
+    private final Processor leftProcessor;
+    private final Processor rightProcessor;
+
+
+    public NullIfProcessor(Processor leftProcessor, Processor rightProcessor) {
+        this.leftProcessor = leftProcessor;
+        this.rightProcessor = rightProcessor;
+    }
+
+    public NullIfProcessor(StreamInput in) throws IOException {
+        leftProcessor = in.readNamedWriteable(Processor.class);
+        rightProcessor = in.readNamedWriteable(Processor.class);
+    }
+
+    @Override
+    public String getWriteableName() {
+        return NAME;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeNamedWriteable(leftProcessor);
+        out.writeNamedWriteable(rightProcessor);
+    }
+
+    @Override
+    public Object process(Object input) {
+        Object leftValue = leftProcessor.process(input);
+        Object rightValue = rightProcessor.process(input);
+        return apply(leftValue, rightValue);
+    }
+
+    public static Object apply(Object leftValue, Object rightValue) {
+        if (BinaryComparisonProcessor.BinaryComparisonOperation.EQ.apply(leftValue, rightValue) == Boolean.TRUE) {
```

Comment:
Maybe `import org.elasticsearch.xpack.sql.expression.predicate.operator.comparison.BinaryComparisonProcessor.BinaryComparisonOperation` to shorten this line and make it more readable?